src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py

```diff
@@ -987,6 +1009,11 @@ def __call__(
                 mask_condition = mask_condition.to(device=device, dtype=masked_image_latents.dtype)
                 condition_kwargs = {"image": init_image_condition, "mask": mask_condition}
             image = self.vae.decode(latents / self.vae.config.scaling_factor, return_dict=False, **condition_kwargs)[0]
+            # If force_unmasked_unchanged, use the original mask in pixel space to recover the original pixels
+            if force_unmasked_unchanged:
+                # Make sure image is on CPU
+                image = image.cpu()
+                image = (1 - mask_condition[:1]) * init_image[:1] + mask_condition[:1] * image
```

Comment:
If this only applies to the output of the pipeline, do we need to add code into source to support this? Might be better suited for a note in the docs on how to use the mask to apply the output image only on the masked area. It's a small code change but anything we can do to keep the loc count down is helpful

Reply:
I have to agree with @williamberman here. I think it is a  small user case and a pretty hacky workaround, so it is better to use the doc ( I didn't realize we can just apply the mask to the output) 

We should still update the tests for inpainting here, though!

Reply:
For adding a note to the docs, would we want the example code to support all of the possible input cases for `image` and `mask_image` (such as `PIL.Image.Image` and `torch.FloatTensor` inputs, batched and unbatched inputs, `mask_images` which aren't binarized, etc.)? In the current code, `mask_condition` and `init_image` above come from the outputs of `prepare_mask_and_masked_image`, which handles a bunch of cases regarding the shapes of `image` and `mask_image`, as well as binarizing the `mask_image`.

One solution could be to to expose a function similar to `prepare_mask_and_masked_image` in `diffusers.utils` (kind of like `diffusers.utils.load_image`) which handles all of the possible inputs, and then use that function in the docs. This might be at odds with the goal of reducing the LOC count, however.

Reply:
I think I'd make the example in the docs focussed on the simplest possible example. Just show how to return a batch 1 tensor and how to mask it. If people want to generalize from there, I think they can figure that out on their own